src/AlertBox/AlertBox.tsx

```diff
@@ -24,10 +33,25 @@ const cssClass = {
   CLOSE: "AlertBox--close",
 };
 
+const TYPES = {
```

Comment:
Any reason for not making this an enum? And then using that enum in the `Props` definition instead of or-ing string literals that have to be manually kept in sync with the values in this object?

```ts
enum Type {
  WARNING = "warning",
  SUCCESS = "success",
  ERROR = "error",
  INFO = "info",
}

export interface Props {
  ...
  type?: Type;
}
```

Reply:
What do you think about this:
```ts
type AlertBoxType = typeof alertBoxTypes;

const alertBoxTypes = ["warning", "success", "error", "info"] as const;
```

That makes `AlertBoxType` of type `"warning" | "success" | "error" | "info"`, and it can be used in the `Props` declaration. 

I think making it an enum would require `Type` to be exported, and all consumers would need to be updated to use the enum. I favor unions of literals over enums. enums are more verbose, and they tend to have keys in all caps which is way too aggressive for me. 
`AlertBox.Type.WARNING` 😨
`"warning"`  😎 
Another drawback to enums is that they emit extra javascript, whereas a union type doesn't emit any javascript.

`Type` is currently only used here
https://github.com/Clever/components/blob/5967a9f8d5f1cfa8d7c7efa1ee739f90ecabd0ae/src/AlertBox/AlertBox.jsx#L82, and it's being converted to an array so there's no reason it needs to be a plain object.

Reply:
Ah great point about having to update consumers if we switch to an enum. Given that we wanna make it as easy as possible to upgrade to the typed version of clever-components, I agree that we should do something more backwards compatible.

-----

In terms of your suggestion, I generally like it! I don't think it's super intuitive that `typeof` `["warning", "success", "error", "info"] as const` will generate a `"warning" | "success" | "error" | "info"` type. But I probably just need to brush up on my TypeScript.

-----

For new components, I think I still lean toward enums for clarity. They just feel like the right tool for the job. But also a valid point about the extra generated JS. This is a question that we can obviously hash out at a later time -- maybe next Frontend Guild meeting!

Reply:
Chatted with @prime-time offline about the merits of enums vs. unions of string literals:

```
enum Size {
  SMALL = "small",
  MEDIUM = "medium",
  LARGE = "large",
}

type Size = "small" | "medium" | "large";
```

I think I've been convinced of the merits of string literal unions! They give us the same type-checking benefits as enums, don't need to be imported by consumers, and are generally lighter weight.

Long-term, we can head in this direction. @prime-time It's also worth bringing this up at the next Frontend Guild meeting!

Reply:
I agree, but I'm also biased 😄